automation_testing/tests/dataview/save_as.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { test, expect } from '@playwright/test';
+import { dataview } from '../../utils/data/dataview_data';
+import { DataviewPage } from '../../pages/DataViewPage';
+import { SaveAsComponent } from '../../pages/SaveAsComponent';
+
+test.describe('Data View', () => {
+
+    let dataviewPage: DataviewPage;
+    let saveAs: SaveAsComponent;
+    test.beforeEach(async ({ page }) => {
+        dataviewPage = new DataviewPage(page);
+        saveAs = dataviewPage.saveAsComponent;
+
+        await dataviewPage.visit();
+    });
+
+ test('Save As', async ({ }) => {
+        await dataviewPage.validateSnatshot(await saveAs.saveAsBtn, 'save_as_btn');
+        await saveAs.saveAsBtn.click();
+        await dataviewPage.validateSnatshot(await saveAs.saveAsOptions, 'save_as_options');
+    });
+
+    test('Save As - New View', async ({ }) => {
+        await saveAs.saveAsBtn.click();
+        await saveAs.selectSaveAsOption(1);
+        await dataviewPage.validateSnatshot(await saveAs.saveView, 'save_as_view');
+        await dataviewPage.validateSnatshot(await saveAs.saveLabel, 'save_as_label');
+        await saveAs.saveLabel.type(dataview.saveAsView);
+        await saveAs.saveViewBtn.click();
+        expect(await saveAs.viewBtn.textContent()).toContain(dataview.saveAsView);
+        await saveAs.viewBtn.click();
+        expect(await (await saveAs.getViewTypeByLabel(dataview.saveAsView)).textContent()).toContain(dataview.viewNotSharedType);
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 2 tabs but found 8 spaces.](https://app.codacy.com/gh/simpleviewinc/sv-mosaic/pullRequest?prid=9498610)
```suggestion
		expect(await (await saveAs.getViewTypeByLabel(dataview.saveAsView)).textContent()).toContain(dataview.viewNotSharedType);
```